### PR TITLE
Feature: documentation and install ordered list 

### PR DESCRIPTION
**Pull request by AlexFernandes-MOVAI**
- Enhance documentation on mobros
- Install ordered list 

----
- [x] Make sure you are opening from a **topic/feature/bugfix branch**
- [x] Ensure that the PR title represents the desired changes
- [x] Ensure that the PR description detail the desired changes
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -66,17 +110,47 @@ Required environment variables:
...
+
+Generated artifacts:
+- Generates a file called "tree.mobtree" where the command was executed, with a resume'd dependency tree of what the packages the user requested.
+
+#### Conflict Reporting <a id="cmd-install-conflict-report"/>
+
+![image](https://user-images.githubusercontent.com/84720623/231483118-44587cbf-3e3f-46fe-9f9c-c1a5329ed1a9.png)
+
+Mobros analyses the dependency tree, and takes in consideration the packages that are installed. The purpose is to have a cautious upgrade of the environment packages, reducing possible impact to a minimum.
+To further troubleshoot where this clash is coming from, its sugested to analyze the tree.mobtree file.
```
Comment (AlexFernandes-MOVAI) [inline README.md:132]: ~~its~~ `it is` or  `it's`

Alex, english teacher

File: README.md [hunk 2]
```diff
@@ -66,17 +110,47 @@ Required environment variables:
...
+
+#### Conflict Handling <a id="cmd-install-conflict-handling"/>
+
+Now that you are facing a conflict in the dependency tree, you need to take a decision and provided it to mobros.
```
Comment (AlexFernandes-MOVAI) [inline README.md:136]: ~~provided~~ provide

File: README.md [hunk 3]
```diff
@@ -66,17 +110,47 @@ Required environment variables:
...
+To understand how to solve you must first understand what mobros expects from the inputs he receives:
+- Package with a version (package_name=package_version): Mobros assumes that the user really wants this specific version, upgrading/downgrading it if it's installed.
```
Comment (AlexFernandes-MOVAI) [inline README.md:138]: maybe be add the fact that the `package_version` needs to be fully correct:
- including build_id
- no wildcard support
- matching the repository name

File: README.md [hunk 4]
```diff
@@ -66,17 +110,47 @@ Required environment variables:
...
+- Package without a version: Mobros considers the installation of the latest version avaiable of the package.
```
Comment (AlexFernandes-MOVAI) [inline README.md:139]: ~~avaiable~~ available

File: README.md [hunk 5]
```diff
@@ -66,17 +110,47 @@ Required environment variables:
  - AWS_SECRET_ACCESS_KEY
 
 
+### Mobros command: install <a id="cmd-install"/>
+
+`Requires escalated privilages`
```
Comment (AlexFernandes-MOVAI) [inline README.md:115]: privilages?

File: README.md [hunk 6]
```diff
@@ -66,17 +110,47 @@ Required environment variables:
...
+
+Example of usage:
+```
+mobros install --pkg_list package_1=0.0.0-0 package_2 package_3=1.1.0-4
```
Comment (AlexFernandes-MOVAI) [inline README.md:119]: can we install local .deb ?

File: README.md [hunk 7]
```diff
@@ -87,7 +161,7 @@ You can manually check the result of the translation by executing:
 rosdep resolve <name_of_dependency>
 ```
 
-#### Ros Metapackages
+#### Ros Metapackages <a id="system-detail-build-ros-metapkg"/>
 
 Another important thing to keep in mind is that, in a project that has a vertical structure, keep in mind that you need to define ros metapackages through the layers for the rosdep to be able to "crawl through" and find all package dependencies it needs to install.
```
Comment (AlexFernandes-MOVAI) [inline README.md:166]: `keep in mind` repeated twice

File: README.md [hunk 8]
```diff
@@ -136,23 +210,13 @@ If you followed the right structure, during mobros packaging you will see the fo
...
+### Installing packages <a id="system-detail-installing"/>
 
-In your project copy and execute the following in your terminal:
-
-```
-wget -qO - https://movai-scripts.s3.amazonaws.com/ros-build.bash | bash
-```
-
-#### Did you know ?
+Motivation: As ros component packaging is debian, it has the necesseties of dependency relationing and dependency treeing as a normal programming language, like mentioning a certain version, even if new versions are avaiable. APT (debian's package manager) has the purpose of keeping operating system packages updated, with the intent of installing latest versions possible, preventing the capability of multi-release lines of a component, or frozen dependencies between ros components.
```
Comment (AlexFernandes-MOVAI) [inline README.md:215]: necesseties ?

File: README.md [hunk 9]
```diff
@@ -136,23 +210,13 @@ If you followed the right structure, during mobros packaging you will see the fo
...
-![Screenshot from 2021-11-22 15-27-02](https://user-images.githubusercontent.com/84720623/142888504-9f480db3-3ed7-428c-838f-924816b1f989.png)
-
-With your new packages referenced there, you only need to :
-
-```
-movai-cli firmware my-robot
-```
+Mobros already does some automatic conflict handling.
+The current automatic conflict resolutions done are:
+-  if the dependency conflict is with an installed version, and their difference is simply in the build identifier (4 digit of the deb package version), it ignores the installed version. The rationale is that build versions are considered security/bugfixes.
```
Comment (AlexFernandes-MOVAI) [inline README.md:222]: aren't there more automatic resolution or mobros then rely on apt ?

**Review by AlexFernandes-MOVAI — COMMENTED** [review]
would be nice to add some links on the subject , like limitation of apt resolver of deps , or rosdep and bloom forums

File: README.md (hunk 9 above)
Comment (duartecoelhomovai) [reply README.md:222]: There is the potential to make it more smart. Right now on mobros side this is as far as it goes.

File: README.md (hunk 6 above)
Comment (duartecoelhomovai) [reply README.md:119]: Not yet

**Comment by duartecoelhomovai** [comment]
Just noticed i commited the ordered list into this branch -_-

**Pull request merged**